AttributeError getsitepackages on Appveyor with Python 2.7 32-bit

On Appveyor with native Windows on Python 2.7 32-bit, we get the following issue since a few days, during `make check`:

```
[376] C:\projects\pywbem$ 'C:\ProgramData\chocolatey\bin\make.EXE' check
Traceback (most recent call last):
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 703, in <module>
    main()
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 683, in main
    paths_in_sys = addsitepackages(paths_in_sys)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 282, in addsitepackages
    addsitedir(sitedir, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 204, in addsitedir
    addpackage(sitedir, name, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 173, in addpackage
    exec(line)
  File "<string>", line 1, in <module>
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site-packages\win32\lib\pywin32_bootstrap.py", line 14, in <module>
    for maybe in site.getsitepackages():
AttributeError: 'module' object has no attribute 'getsitepackages'
Traceback (most recent call last):
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 703, in <module>
    main()
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 683, in main
    paths_in_sys = addsitepackages(paths_in_sys)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 282, in addsitepackages
    addsitedir(sitedir, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 204, in addsitedir
    addpackage(sitedir, name, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 173, in addpackage
    exec(line)
  File "<string>", line 1, in <module>
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site-packages\win32\lib\pywin32_bootstrap.py", line 14, in <module>
    for maybe in site.getsitepackages():
AttributeError: 'module' object has no attribute 'getsitepackages'
Traceback (most recent call last):
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 703, in <module>
    main()
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 683, in main
    paths_in_sys = addsitepackages(paths_in_sys)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 282, in addsitepackages
    addsitedir(sitedir, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 204, in addsitedir
    addpackage(sitedir, name, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 173, in addpackage
    exec(line)
  File "<string>", line 1, in <module>
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site-packages\win32\lib\pywin32_bootstrap.py", line 14, in <module>
    for maybe in site.getsitepackages():
AttributeError: 'module' object has no attribute 'getsitepackages'
Traceback (most recent call last):
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 703, in <module>
    main()
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 683, in main
    paths_in_sys = addsitepackages(paths_in_sys)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 282, in addsitepackages
    addsitedir(sitedir, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 204, in addsitedir
    addpackage(sitedir, name, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 173, in addpackage
    exec(line)
  File "<string>", line 1, in <module>
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site-packages\win32\lib\pywin32_bootstrap.py", line 14, in <module>
    for maybe in site.getsitepackages():
AttributeError: 'module' object has no attribute 'getsitepackages'
del /f /q pip_upgrade_py.done
Could Not Find C:\projects\pywbem\pip_upgrade_py.done
python -m pip install --upgrade pip
Traceback (most recent call last):
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 703, in <module>
    main()
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 683, in main
    paths_in_sys = addsitepackages(paths_in_sys)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 282, in addsitepackages
    addsitedir(sitedir, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 204, in addsitedir
    addpackage(sitedir, name, known_paths)
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site.py", line 173, in addpackage
    exec(line)
  File "<string>", line 1, in <module>
  File "C:\projects\pywbem\.tox\win64_py27_32\lib\site-packages\win32\lib\pywin32_bootstrap.py", line 14, in <module>
    for maybe in site.getsitepackages():
AttributeError: 'module' object has no attribute 'getsitepackages'
make: *** [makefile:383: pip_upgrade_py.done] Error 1
```

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -69,6 +69,10 @@ twine>=1.8.1
 # Python 3.4.
 jupyter>=1.0.0; python_version != '3.4' or sys_platform != 'win32'
 
+# Pywin32 is used (at least?) by jupyter.
+# TODO: Workaround for issue #1946 is to pin pywin32 to version 225.
+pywin32>=222,<=225; sys_platform == 'win32'
+
 # The tornado package is used by ipykernel which is used by jupyter.
 # Tornado 5.0.0 and 5.0.1 rejects installation if the Python ssl module
 # does not have certain symbols required by Tornado. This issue exists for

--- a/docs/changes.rst
+++ b/docs/changes.rst
@@ -221,6 +221,9 @@ Released: not yet
 
 * Docs: Fixed errors in description of CIMInstance.update_existing().
 
+* Added dependency to pywin32 package for Windows, and pinned it to version 225
+  to work around an issue in its version 226. (See issue ##1946)
+
 **Enhancements:**
 
 * Changed GetCentralInstances methodology in WBEMServer.get_central_instances()

--- a/minimum-constraints.txt
+++ b/minimum-constraints.txt
@@ -96,6 +96,9 @@ twine==1.8.1
 # Jupyter Notebook (no imports, invoked via jupyter script):
 jupyter==1.0.0
 
+# Pywin32 is used (at least?) by jupyter.
+pywin32==222; sys_platform == 'win32'
+
 # Table output formatter used by the manual performance tests to display
 # timing results
 tabulate==0.8.3

